src/System.Management.Automation/engine/CommandCompletion/CompletionAnalysis.cs

```diff
@@ -473,6 +473,13 @@ internal List<CompletionResult> GetResultHelper(CompletionContext completionCont
                         replacementLength = 0;
                         break;
 
+                    case TokenKind.Semi:
+                        // Handle scenarios such as 'gci | Format-Table @{Label=...;<tab>'
+                        result = GetResultForHashtable(completionContext);
```

Comment:
`get-something;<tab>` does go through this path. It's better to check `if (lastAst is HashtableAst)` before calling `GetResultForHashtable`.

Reply:
Fixed.